sale_margin_delivered/models/sale_margin.py

```diff
@@ -45,7 +45,7 @@ def _compute_margin_delivered(self):
                 cost_price += move.product_qty * abs(move.price_unit)
             qty = delivered_qty or line.product_uom_qty
             average_price = qty and (
-                cost_price or line.purchase_price) / qty or 0.0
+                cost_price and qty and cost_price / qty or line.purchase_price) or 0.0
```

Comment:
You can leave it just as: `cost_price / qty or line.purchase_price) or 0.0`

Because:

- `qty` is already checked in line 47.
- if `cost_price` is 0, `cost_price / qty` will be 0 also.